modules/applications/pages/ui-builder/how-to-upload-multiple-documents.adoc

```diff
@@ -0,0 +1,91 @@
+= Upload multiple documents simultaneously
+:page-aliases: applications:how-to-upload-multiple-documents.adoc
+:description: Learn how to upload multiple documents simultaneously
+
+{description}
+
+* Difficulty: intermediate
+* Prerequisites: have an xref:create-an-interface.adoc[interface created] on Bonita UI Builder
+
+If you need to upload multiple files using Bonita's formFileUpload servlet, you'll need to process each file separately by looping through them, adding a parameter to the API query for each file. Since Bonita's `formFileUpload` doesn't support uploading multiple files at once, you'll also need to synchronize the uploads.
+
+
+== Example
+
+Suppose you want to upload multiple files at the same time using a `FilePicker` widget in the Bonita UI Builder.
+
+=== 1. Use a FilePicker widget
+To do so, drag and drop a `FilePicker` widget onto your interface and configure it as follows:
+
+image:ui-builder/guides/file-picker-widget-properties.png[file-picker-widget-propeties]
+
+[TIP]
+In this example, we don't set a limit for the `Max no. of files` because we're allowing multiple files to be uploaded. However, you can set a limit on the number of files to be uploaded, as well as adjust other properties if needed.
+
+
+=== 2. Create a post request
+* Go to `Queries` section and create a new query/API and configure it as follows:
+
+image:ui-builder/guides/upload-files-query.png[upload-files-query]
+
+
+=== 3. Create JavaScript object
+
+* Go to `JS` section and create a new JS object to create functions given below that allows to upload multiple files simultaneously.
+
+* Function to get files
+[source, JS]
+----
+export default {
```

Comment:
**question:** i saw that this script is not colored.
not like the next one
![image](https://github.com/user-attachments/assets/483648de-3ced-4f44-b236-4af3f5d443ab)
![Screenshot 2024-08-21 at 16 40 02](https://github.com/user-attachments/assets/c93cd288-9611-4e5f-8c43-9e006db08f64)